progress.go

```diff
@@ -138,6 +138,17 @@ func (p *progress) close() error {
 	return nil // Either not empty or error, suits both cases
 }
 
+// calculates the number of bytes downloaded
+func (p *progress) downloadedBytes() int64 {
+	var downloaded int64
+	for idx := range p.Chunks {
+		if p.Chunks[idx] == 1 {
+			downloaded += p.ChunkSize
```

Comment:
This is not precise in the case of the last chunk. Let's say a file is 10 bytes long, and the chunk size is 4. The third chunk would be 2 bytes long, not 4.

Reply:
That means we need to track the number of bytes downloaded too. Right?

Reply:
My suggestion is to write the number of bytes downloaded per chunk, instead of simply 0 or 1. What do you think?

Reply:
Good idea!

Reply:
On it.